src/freenet/pluginmanager/PluginManager.java

```diff
@@ -324,12 +308,14 @@ private static String pluginList(ArrayList<PluginInfoWrapper> wrappers) {
 	}
 
 	private String[] getConfigLoadString() {
-		ArrayList<String> v = new ArrayList<String>();
+		List<String> v = new ArrayList<String>();
+		for (PluginInfoWrapper pi : pluginWrappers) {
```

Comment:
Doing it this way has the potential (however unlikely) to return a wrong result due to a race condition where the snapshots of `pluginWrappers` and `pluginsFailedLoad`, not taken atomically as a pair, are inconsistent.

Including a little extra work in the `synchronized` block is not that bad, we don't expect this method to be called too often anyhow.